src/common/number/format_number.ts

```diff
@@ -90,8 +91,15 @@ export const formatNumber = (
  * @returns An `Intl.NumberFormatOptions` object with `maximumFractionDigits` set to 0, or `undefined`
  */
 export const getNumberFormatOptions = (
-  entityState: HassEntity
+  entityState: HassEntity,
+  entity: EntityRegistryEntry | undefined
 ): Intl.NumberFormatOptions | undefined => {
+  const display_precision =
+    entity?.options?.sensor?.display_precision ||
+    entity?.options?.sensor?.suggested_display_precision;
```

Comment:
```suggestion
    entity?.options?.sensor?.display_precision ??
    entity?.options?.sensor?.suggested_display_precision;
```
We must check for `null` and `undefined` not falsy.